task_spec/spec/acceptance/init_spec.rb

```diff
@@ -1,203 +1,45 @@
-          groups << group
-        end
-        group[:nodes] << node
-      else
-        puts "Invalid group name #{group_name} skipping"
-      end
-    end
-
-    nodes = hosts.map do |host|
-      if host[:platform] =~ /windows/
-        config = {
-          transport: 'winrm',
-          winrm: {
-            user: host[:user],
-            password: ENV['BEAKER_password'],
-            ssl: false
-          }
-        }
-        node = {
-          name: host.hostname,
-          config: config
-        }
-      else
-        ssh = host[:ssh]
-        if ssh
-          config = { transport: 'ssh' , ssh: {} }
-          [:password, :user, :port].each { |k| config[:ssh][k] = ssh[k] if ssh[k] }
-
-          # This is required to find the key that may be loaded from config
-          # The vagrant provisioner passes the vagrant ssh config instead of
-          # specific options.
-          key = nil
-          keys = host.connection.instance_variable_get(:@ssh).options[:keys]
-          key = keys.first if keys
-          config[:ssh][:"private-key"] = key if key
-
-          # If the hypevisor stores IP as part of its core config then hostname
-          # is unlikely to resolve in beakers world, but there may be many hosts
-          # on the same IP. Use a query argument to generate a unique hostname.
-          # BOLT-510
-          hostname = host.host_hash[:ip] ? "#{host.host_hash[:ip]}?n=#{host.hostname}" : host.hostname
-          node_name = "ssh://#{hostname}"
-
-          node = {
-            name: node_name,
-            config: config
-          }
-        else
-          raise 'non-ssh targets not yet implemented'
-        end
-      end
-      # Make alias groups for each role
-      host[:roles].each do |role|
-        add_node(node[:name], role, groups)
-      end
-      node
-    end
-    { nodes: nodes,
-      groups: groups,
-      config: {
-        ssh: {
-          'host-key-check' => false
-        }
-      }
-    }
-  end
-
-  def run_task_on(target, task, **opts)
-    result = bolt_on(target, 'task', 'run', task, opts)
-    unless opts[:expect_success] == false
-      expect_node_success(result)
-    end
-    result["items"]
-  end
-
-  def run_command_on(target, command, **opts)
-    result = bolt_on(target, 'command', 'run', command, opts)
-    unless opts[:expect_success] == false
-      expect_node_success(result)
-    end
-    result["items"]
-  end
-
-  def expect_node_success(result)
-    result['items'].each do |res|
-      unless res['status'] == 'success'
-        # We really need matchers for this. rspec is overabstracted
-        useful_info = "#{res.dig('result', '_error','msg')} :\n #{res['result']['_output']}"
-        expect(useful_info).to eq('')
-      else
-        expect(res['status']).to eq('success')
-      end
-    end
-
-  end
+  include Beaker::TaskHelper::Inventory
+  include BoltSpec::Run
 
   def module_path
     RSpec.configuration.module_path
```

Comment:
This feels like something we should maybe put in a helper?

Reply:
This is rspec specific so it doesn't seem like it belongs in beaker-task_helpers. I'm not sure it's the best way to track this either.